### PR TITLE
UIKIT-55: select inputs line between items

### PR DESCRIPTION
**Pull request by vidurar**
#### Summary

add a `shouldDivideOptionGroups` prop to `SelectInput`,`CreatableSelectInput`, `AsyncSelectInput` and `AsyncCreatableSelectInput`, to display a divider between option groups

### CODE REVIEW

File: materials/custom-properties.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ export default {
   borderRadiusInput: '6px',
   borderRadiusTag: '2px',
   borderRadiusForInput: '6px',
+  borderTopForOptionGroupSeparator: '1px solid #ccc',
```
Comment (montezume) [inline materials/custom-properties.js]: this file is custom generated, so you can't add this here as it will get overridden. Maybe just use the custom property ${colorGray} and don't bother with adding a token

Comment (vidurar) [reply materials/custom-properties.js]: 👍

File: src/components/inputs/async-select-input/async-select-input.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ export class AsyncSelectInput extends React.Component {
     ]),
     loadOptions: PropTypes.func.isRequired,
     cacheOptions: PropTypes.any,
+    shouldDivideOptionGroups: PropTypes.bool,
```
Comment (montezume) [inline src/components/inputs/async-select-input/async-select-input.js]: I'm not sure about this prop maybe. Maybe `showOptionGroupDivider` would be more appropriate 🤔

File: src/components/internals/create-select-styles.js [hunk 3]
```diff
@@ -157,22 +157,27 @@ const multiValueRemoveStyles = () => base => ({
...
-    borderTop: `1px ${vars.borderColorSeparator} solid`,
+  '&:not(:first-child)': {
+    borderTop: props.shouldDivideOptionGroups
+      ? `1px solid ${vars.colorGray}`
+      : base.borderTop,
   },
 });
 
-const groupHeadingStyles = () => base => ({
+export const groupHeadingStyles = () => base => ({
```
Comment (montezume) [inline src/components/internals/create-select-styles.js]: Why this change?

Comment (vidurar) [reply src/components/internals/create-select-styles.js]: this was something I added while figuring out the implementation and forgot to remove. Will get rid of this

File: src/components/inputs/async-select-input/async-select-input.js (hunk 2 above)
Comment (qmateub) [reply src/components/inputs/async-select-input/async-select-input.js]: To me `divide` is already an action, so maybe directly `divideOptionGroups` is enough 😅

Comment (vidurar) [reply src/components/inputs/async-select-input/async-select-input.js]: i think `showOptionGroupDivider` is a little clearer

**Comment by montezume** [comment]
Just need to modify those VRT routes so that we have 3 snapshot-ed states for the open select input and we should be good to go 👍 

- Open without option groups
- Open with option groups
- Open with option groups and `showOptionGroupDivider` enabled

**Comment by montezume** [comment]
Okay I tagged @mariabarrena to review the Percy VRTs.

File: src/components/internals/create-select-styles.js [hunk 4]
```diff
@@ -157,11 +157,13 @@ const multiValueRemoveStyles = () => base => ({
...
-const groupStyles = () => base => ({
+const groupStyles = props => base => ({
   ...base,
   padding: 0,
-  '&:not(:firstChild)': {
-    borderTop: `1px ${vars.borderColorSeparator} solid`,
+  '&:not(:first-child)': {
+    borderTop: props.showOptionGroupDivider
+      ? `1px solid ${vars.colorGray}`
+      : base.borderTop,
```
Comment (emmenko) [inline src/components/internals/create-select-styles.js]: Is this correct? Where does it come from? 🤔

Comment (montezume) [reply src/components/internals/create-select-styles.js]: base.borderTop is `undefined`. I think this is what we want?

Comment (vidurar) [reply src/components/internals/create-select-styles.js]: This value is undefined , but used `base.borderTop` instead of hardcoding `undefined` to stick to the patterns found elsewhere in this file

Comment (tdeekens) [reply src/components/internals/create-select-styles.js]: This seems closed over from [here](https://github.com/commercetools/ui-kit/blob/47d5de1e53f5dca1a569bcd8f1158bbe10ace967/src/components/internals/create-select-styles.js#L195) with the `props` [here](https://github.com/commercetools/ui-kit/blob/7c53c7d3b301a3930917c8456eea2f87d7c446e0/src/components/inputs/select-input/select-input.js#L175). Then I fail to follow from react-select what `base is. Should that be our `vars`?

Comment (vidurar) [reply src/components/internals/create-select-styles.js]: From when i looked into it, `base` are the default styles for the component provided by `react-select` itself.

Comment (montezume) [reply src/components/internals/create-select-styles.js]: Yeah, I think it's fine to use base.borderTop. We are doing similar things in the rest of these functions.

**Comment by montezume** [comment]
@emmenko do you have any more concerns about this or is it okay to merge from your side?
